src/Whoops/Exception/Inspector.php

```diff
@@ -30,12 +32,19 @@ class Inspector
      */
     private $previousExceptions;
 
+    /**
+     * @var \Whoops\Inspector\InspectorFactoryInterface|null
+     */
+    protected $inspectorFactory;
+
     /**
      * @param \Throwable $exception The exception to inspect
+     * @param \Whoops\Inspector\InspectorFactoryInterface $factory
      */
-    public function __construct($exception)
+    public function __construct($exception, $factory = null)
```

Comment:
Do we have to allow this override? I’d prefer this if we can:

```suggestion
    public function __construct($exception)
```

Reply:
Hey thanks for taking a look! Really appreciated.

I agree it feels a bit weird. I made it default null because I didn't want to break backwards compat should anyone be using the inspector outside of this project. It's also required because the instance needs to be retained for when new inspectors are created, such as `getPreviouslyExceptionInspector()`. 

That all said, there might be a better way? Maybe just a `setInspectorFactory()` method so we're not polluting the constructor.